core/templates/dev/head/filters.js

```diff
@@ -404,3 +404,34 @@ oppia.filter('stripFormatting', [function() {
     return strippedText;
   };
 }]);
+
+/* The following filter replaces each RTE element occurrence in the input html
+   by its corresponding name in square brackets and returns a string
+   which contains the name in the same location as in the input html.
+   eg: <p>Sample1 <oppia-noninteractive-math></oppia-noninteractive-math>
+        Sample2 </p>
+   will give as output: Sample1 [Math] Sample2 */
+oppia.filter('formatRtePreview', ['$filter', function($filter) {
+  return function(html) {
+    html = html.replace(/&nbsp;/ig, ' ');
+    html = html.replace(/&quot;/ig, '');
+    //Replace all html tags other than <oppia-noninteractive-**> ones to ''
+    html = html.replace(/<(?!oppia-noninteractive\s*?)[^>]+>/g, '');
+    html = html.split(/[<>]+/);
```

Comment:
This varname should now be htmlArray, right? (I was looking at the subsequent code and thinking that html[i] was a character...)

Reply:
Done, removed the use of htmlArray.